e2e-tests/src/main.rs

```diff
@@ -248,3 +276,68 @@ fn accounts(seeds: Option<Vec<String>>) -> Vec<sr25519::Pair> {
     });
     seeds.into_iter().map(keypair_from_string).collect()
 }
+
+fn get_first_two_accounts(accounts: &[sr25519::Pair]) -> (sr25519::Pair, sr25519::Pair) {
+    let first = accounts.get(0).expect("No accounts passed").to_owned();
+    let second = accounts
+        .get(1)
+        .expect("Pass at least two accounts")
+        .to_owned();
+    (first, second)
+}
+
+#[derive(Debug)]
+struct FeeInfo {
+    base_fee: Balance,
+    len_fee: Balance,
+    unadjusted_weight: Balance,
+    adjusted_weight: Balance,
+    tip: Balance,
+}
+
+fn get_tx_fee_info(connection: &Connection, tx: &TransferTransaction) -> FeeInfo {
+    let block = connection.get_block::<Block>(None).unwrap().unwrap();
+    let block_hash = block.header.hash();
+
+    let unadjusted_weight = connection
+        .get_payment_info(&tx.hex_encode(), Some(block_hash))
+        .unwrap()
+        .unwrap()
+        .weight as Balance;
```

Comment:
looks suspicious  :raised_eyebrow:

Reply:
well, using the exact type `Weight` requires adding new dependency, so I chose 'suspicious' cast; I guess that in tests it is acceptable somehow, especially since effectively we use weight as balance

Reply:
ok